toolkit/tools/scheduler/scheduler.go

```diff
@@ -346,6 +366,8 @@ func buildAllNodes(stopOnFailure, canUseCache bool, packagesToRebuild, testsToRe
 		if len(buildState.ActiveBuilds()) == 0 && len(channels.Results) == 0 {
 			if useCachedImplicit {
 				err = fmt.Errorf("could not build all packages")
+				// Temporarily print debug information about the stuck node.
+				debugStuckNode(buildState, pkgGraph, goalNode, 0)
```

Comment:
I'd wrap this function in another to print an initial message saying we're printing a stack of nodes, on which node `goalNode` was stuck on. This way we won't have to keep printing "STUCK" on each line, because it's going to be clear from that initial message.